pkg/prometheus/promcfg.go

```diff
@@ -1611,16 +1611,17 @@ func generateRunningFilter() yaml.MapSlice {
 	}
 }
 
-func getLimit(user *uint64, enforced *uint64) *uint64 {
+func (cg *ConfigGenerator) getLimit(user *uint64, enforced *uint64) *uint64 {
 	if enforced == nil {
 		return user
 	}
 
-	if user == nil {
+	// If the Prometheus version doesn't support global limit
```

Comment:
Would it be easier to follow with this logic?
* if *user == 0
  * if version >= 2.45.0, return nil (global limit applies)
  * otherwise return enforced